src/components/Button/Button.tsx

```diff
@@ -31,33 +31,34 @@ export const Button = forwardRef<HTMLButtonElement, ButtonProps>(
   function Button(
     {
       text,
-      type,
-      size,
+      type = Type.PRIMARY,
+      buttonType,
+      size = Size.MEDIUM,
       leftIcon,
       rightIcon,
       isDisabled,
-      includeHoverAnimations,
+      disabled: disabledProp,
+      includeHoverAnimations = true,
       widthToHideText,
       minWidth,
       className,
       iconClassName,
       textClassName,
-      name,
-      onClick,
-      onMouseEnter,
-      onMouseLeave,
-      onMouseUp,
-      onMouseDown
+      'aria-label': ariaLabelProp,
+      ...other
     },
     ref
   ) {
     const { textIsHidden } = useCollapsibleText(widthToHideText)
+    const disabled = disabledProp ?? isDisabled
+    const noText = !text || textIsHidden
+    const hasText = !!text && !textIsHidden
```

Comment:
Isn't this just the inverse of the above? Would make it less confusing to just use `!hasText` instead of `noText` imo

Reply:
yeah def